### DISCUSSION TITLE
Combo Tap Dance?

### DISCUSSION THREAD

**Discussion by c2keesey** [urob/zmk-config#63] [Q&A] [answered]
Hi, I'd like to create a combo tap dance to trigger &kp LG(A) on first combo tap, and &kp LG(C) on second. Does anyone know how to implement this behavior? Here is my attempt, but it is not working.
```
// copy all macro
ZMK_BEHAVIOR(copy_all, macro,
    wait-ms = <15>;
    tap-ms = <15>;
    bindings = <&kp LG(A)>, <&kp LG(C)>;
)

ZMK_BEHAVIOR(copy_all_td, tap_dance,
    tapping-term-ms = <200>;
    bindings = <&kp LG(A)>, <&copy_all>;
)
MAKE_HRM(hml_copy_all, &kp, &copy_all_td, KEYS_R THUMBS)  // "    "   "   "

#define HRC_CP_ALL &hml_copy_all LA(LSHFT) 0
ZMK_COMBO(copy_all, HRC_CP_ALL, LM1 LM3,     DEF NAV NUM, COMBO_TERM_SLOW)
```

**Comment by urob** [comment, marked as answer]
> Hi, I'd like to create a combo tap dance to trigger `&kp LG(A)` on first combo tap, and `&kp LG(C)` on second. 

You don't need the `macro`, `HRM` or `combo` for that. The following should do:
```c
ZMK_BEHAVIOR(copy_all_td, tap_dance,
    tapping-term-ms = <200>;
    bindings = <&kp LG(A)>, <&kp LG(C)>;
)
```
You can then add `&copy_all_td` anywhere on your keymap or bind it to other behaviors such as combos or hold-taps.

**Reply by urob** [reply]
Btw, if you are using the `v2` branch of `zmk-helpers`, this can be shortened to:
```c
ZMK_TAP_DANCE(copy_all_td,
    tapping-term-ms = <200>;
    bindings = <&kp LG(A)>, <&kp LG(C)>;
)
```

**Reply by c2keesey** [reply]
Thank you so much! Been loving the setup for my corne. Really appreciate your repo!!
